### PR TITLE
KCB-1088-cannot-add-listApp-grid

### PR DESCRIPTION
**Pull request by Pham-Gia-Huong**

### CODE REVIEW

File: src/renderer/Layout/GridLayout/GridBlock/renderer.tsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ const buildContent = ({content, type = CONTENT_TYPE.EMPTY as ContentType}:
     case CONTENT_TYPE.APP_SPACE: {
       const blockContentAppSpace = content as AppSpaceWidgetProps;
       currentContentBlock = (
-        <AppSpaceWidget widgetTitle={blockContentAppSpace.widgetTitle} listContent={blockContentAppSpace.listContent} />
+        <AppSpaceWidget widgetTitle={blockContentAppSpace.widgetTitle} contentList={blockContentAppSpace.contentList} />
```
Comment (phongnhat19) [inline src/renderer/Layout/GridLayout/GridBlock/renderer.tsx:70]: This change is not related to bug KCB-1088

File: src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx [hunk 2]
```diff
@@ -9,11 +9,11 @@ const CategoryDetail = ({
   onChangeRow,
 }: {
   categoryList: any;
-  onChangeRow?: (categoryList: Array<{id: string; type: string}>) => void;
+  onChangeRow?: (categoryList: Array<{id: string; type: string; name: string; icon: string}>) => void;
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx:12]: This change is not related to bug KCB-1088

File: src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx [hunk 3]
```diff
@@ -9,11 +9,11 @@ const CategoryDetail = ({
...
 }) => {
   const newCategoryList = categoryList.slice();
   const addRow = (i: number) => {
-    newCategoryList.splice(i + 1, 0, {type: 'app', id: ''});
+    newCategoryList.splice(i + 1, 0, {type: 'app', id: '', name: '', icon: ''});
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx:16]: This change is not related to bug KCB-1088

File: src/renderer/Widget/AppSpaceListWidget/type.d.ts [hunk 4]
```diff
@@ -13,5 +13,5 @@ declare type CategorytAppSpace = {
 
 declare type ModalAppSpaceContent = {
   category: string;
-  categoryList: ContentAppSpaceWidget[];
+  categoryList: CategorytAppSpace[];
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/type.d.ts:16]: This change is not related to bug KCB-1088

File: src/renderer/Widget/AppSpaceListWidget/service.ts [hunk 5]
```diff
@@ -24,7 +24,7 @@ const getAppInfo: ({listAppSpace}: {listAppSpace: ModalAppSpaceContent[]}) => Pr
       }
       let appInfo = {} as {name: string; icon: string; message: any};
       try {
-        appInfo = await getApp({id: category.id});
+        appInfo = await getApp({id: Number(category.id)});
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/service.ts]: This change is not related to bug KCB-1088

File: src/renderer/Widget/AppSpaceListWidget/service.ts [hunk 6]
```diff
@@ -5,15 +5,15 @@ const getAppInfo: ({listAppSpace}: {listAppSpace: ModalAppSpaceContent[]}) => Pr
...
+  const newListAppSpace = listAppSpace.map(appSpace => JSON.parse(JSON.stringify(appSpace)));
   for (let i = 0; i < newListAppSpace.length; i++) {
     const appSpace = newListAppSpace[i];
     for (let j = 0; j < appSpace.categoryList.length; j++) {
       const category = appSpace.categoryList[j];
       if (category.type === 'space') {
         let spaceInfo = {} as any;
         try {
-          spaceInfo = await getSpace({id: category.id});
+          spaceInfo = await getSpace({id: Number(category.id)});
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/service.ts]: This change is not related to bug KCB-1088

File: src/renderer/Widget/AppSpaceListWidget/service.ts [hunk 7]
```diff
@@ -5,15 +5,15 @@ const getAppInfo: ({listAppSpace}: {listAppSpace: ModalAppSpaceContent[]}) => Pr
 }: {
   listAppSpace: ModalAppSpaceContent[];
 }) => {
-  const newListAppSpace = listAppSpace.slice();
+  const newListAppSpace = listAppSpace.map(appSpace => JSON.parse(JSON.stringify(appSpace)));
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/service.ts:8]: This change is not related to bug KCB-1088

File: src/renderer/Widget/AppSpaceListWidget/renderer.tsx [hunk 8]
```diff
@@ -7,14 +7,59 @@ import Note from './components/Note';
...
+            && categoryUpdate.id === category.id
+            && categoryUpdate.icon === category.icon
+        );
+        if (!isSameCategory) {
+          return false;
+        }
+      }
+      return true;
+    };
+    const isSameCategoryValue = (currentObjectList: ModalAppSpaceContent[], updateObjectList: ModalAppSpaceContent[]) => {
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/renderer.tsx]: Separate `isSameCategoryValue` and `isSameCategoryObject` into `util` file

**Review by phongnhat19 — CHANGES_REQUESTED** [review]

File: src/renderer/Layout/GridLayout/GridBlock/renderer.tsx (hunk 1 above)
Comment (Pham-Gia-Huong) [reply src/renderer/Layout/GridLayout/GridBlock/renderer.tsx:70]: due listContent in component already change to contentList.
If not change params pass into component will not work.

File: src/renderer/Widget/AppSpaceListWidget/type.d.ts (hunk 4 above)
Comment (Pham-Gia-Huong) [reply src/renderer/Widget/AppSpaceListWidget/type.d.ts:16]: the change related to new function isSameCategoryValue and isSameCategoryValue

File: src/renderer/Widget/AppSpaceListWidget/service.ts (hunk 5 above)
Comment (Pham-Gia-Huong) [reply src/renderer/Widget/AppSpaceListWidget/service.ts:27]: the change related to new function isSameCategoryValue and isSameCategoryValue

File: src/renderer/Widget/AppSpaceListWidget/service.ts (hunk 7 above)
Comment (Pham-Gia-Huong) [reply src/renderer/Widget/AppSpaceListWidget/service.ts:8]: the change related to new function isSameCategoryValue and isSameCategoryValue

File: src/renderer/Widget/AppSpaceListWidget/renderer.tsx (hunk 8 above)
Comment (Pham-Gia-Huong) [reply src/renderer/Widget/AppSpaceListWidget/renderer.tsx]: isSameCategoryValue and isSameCategoryObject must set in context of useEffect.
Separate these function into util file, eslint will show warning.

File: src/renderer/Widget/AppSpaceListWidget/service.ts (hunk 6 above)
Comment (Pham-Gia-Huong) [reply src/renderer/Widget/AppSpaceListWidget/service.ts]: resolve

File: src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx (hunk 2 above)
Comment (Pham-Gia-Huong) [reply src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx:12]: the change related to new function isSameCategoryValue and isSameCategoryValue

File: src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx (hunk 3 above)
Comment (Pham-Gia-Huong) [reply src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx:16]: the change related to new function isSameCategoryValue and isSameCategoryValue

File: src/renderer/Widget/AppSpaceListWidget/service.ts (hunk 5 above)
Comment (Pham-Gia-Huong) [reply src/renderer/Widget/AppSpaceListWidget/service.ts]: resolve

File: src/renderer/Widget/AppSpaceListWidget/renderer.tsx (hunk 8 above)
Comment (Pham-Gia-Huong) [reply src/renderer/Widget/AppSpaceListWidget/renderer.tsx]: resolve

**Review by phongnhat19 — APPROVED** [review]
